File: nyx_fuzzer/structured_fuzzer/interpreter/spec_lib/generators.py
class FixedOptionsGenerator:
    def __init__(self,*options):
        self.options=options
    
    def msgpack(self):
        return ["Options",self.options]

def opts(*options):
    return FixedOptionsGenerator(*options)

class FixedFlagsGenerator:
    def __init__(self,*options):
        self.flags=options
    
    def msgpack(self):
        return ["Flags",self.flags]

def flags(*options):
    return FixedOptionsGenerator(*options)

class LimitsGenerator:
    def __init__(self,min,max,align):
        self.min=min
        self.max=max
        self.align = align
    
    def msgpack(self):
        return ["Limits",(self.min,self.max),self.align]

def limits(min,max,align=1):
    return LimitsGenerator(min,max,align)

File: nyx_fuzzer/structured_fuzzer/src/graph_mutator/generators.rs
use crate::random::distributions::Distributions;

#[derive(Debug, PartialEq, Deserialize, Serialize, Clone)]
#[serde(tag = "type")]
pub enum IntGenerator{
    Options{opts: Vec<u64>},
    Flags{opts: Vec<u64>},
    Limits{range:(u64, u64), align: u64},
}

impl IntGenerator{
    pub fn generate(&self, dist: &Distributions) -> (u64,bool){
        use IntGenerator::*;
        match self {
            Options{opts} => (opts[dist.gen_range(0,opts.len())],false),
            Flags{opts} => (opts[dist.gen_range(0,opts.len())],false),
            Limits{range,align} => {
                let mut val = dist.gen_range(range.0,range.1);
                val = val-(val%align);
                if val < range.0 {val+=align}
                if val > range.1 {val-=align}
                (val,false)
            },
        }
    }
}